frontend/src/Game.js

```diff
@@ -248,37 +248,35 @@ export default class Game {
             }
 
             // check all enemies, attack first (farthest along path)
-            let towerAttacked = false;
             for (let enemy of this.enemies) {
-                if (!enemy.hp) continue;
-                const path = tower.getProjectilePath(enemy.position, this.gameMap);
-
-                if (path.length) {
-                    if (!towerAttacked) { // prevent stacked projectiles hitting same tile
-                        if (tower.appliedStatus) {
-                            if (enemy.statuses.has(tower.appliedStatus)) {
-                                continue;
-                            }
-                            enemy.statuses.add(tower.appliedStatus);
-                        }
-                        // TODO: this is instant damage, convert to when projectile hits?
-                        if (tower.damage) {
-                            let damage = tower.buffs.has(BUFFED) ? tower.damage * 2 : tower.damage;
-                            damage = Math.min(enemy.hp, damage)
-
-                            this.damage_dealt += damage
-                            enemy.hp = enemy.hp - damage
-                        }
-
-                        const projectile = tower.createProjectile(path);
-                        this.addProjectile(projectile);
+                if (!enemy.hp) continue; // skip enemy is dead
+
+                const travelTime = tower.getTravelTime(enemy.position, this.gameMap);
+                const futurePosition = enemy.getFutureLocation(travelTime);
+                if (!futurePosition) continue; // skip if enemy will reach castle
+
+                const path = tower.getProjectilePath(futurePosition, this.gameMap, travelTime);
+                if (!path) continue; // skip if tower can't hit enemy
+
+                if (tower.appliedStatus) { // handle status towers
+                    if (enemy.statuses.has(tower.appliedStatus)) {
+                        continue;
                     }
+                    enemy.statuses.add(tower.appliedStatus);
+                }
 
-                    towerAttacked = true;
-                    if (!tower.canAttackMultiple) break;
+                // TODO: this is instant damage, convert to when projectile hits?
+                if (tower.damage) {
+                    const damage = tower.buffs.has(BUFFED) ? tower.damage * 2 : tower.damage;
+                    enemy.hp = Math.max(enemy.hp - damage, 0);
```

Comment:
Is it just me or did this not merge correctly? Damage tracking seems to be missing